front/src/applications/operationalStudies/components/ImportTrainSchedule/ImportTrainScheduleModal.jsx

```diff
@@ -123,83 +124,77 @@ export default function ImportTrainScheduleModal(props) {
   // 2 GENERATE PATHS (autocomplete to automatically look for OPs)
   //
 
-  async function launchPathfinding(
-    params,
-    pathRefNum,
-    pathNumberToComplete,
-    pathsIDs,
-    continuePath,
-    autoComplete
-  ) {
-    try {
-      const itineraryCreated = autoComplete
-        ? await postPathFindingOp({ pathOpQuery: params }).unwrap()
-        : await postPathFinding({ pathQuery: params }).unwrap();
-
-      continuePath(
-        pathNumberToComplete + 1,
-        {
-          ...pathsIDs,
-          [pathRefNum]: {
-            pathId: itineraryCreated.id,
-            rollingStockId: params.rolling_stocks[0],
-            pathFinding: itineraryCreated,
-          },
-        },
-        autoComplete
+  function endGeneratePaths(newTrainsWithPathRef) {
+    if (newTrainsWithPathRef.length > 0) {
+      setImportStatus(t('operationalStudies/importTrainSchedule:status.pathComplete'));
+    } else setImportStatus(t('operationalStudies/importTrainSchedule:status.pathsFailed'));
+    setTrainsWithPathRef(newTrainsWithPathRef);
+    setStatus({ ...status, uicComplete: true, pathFindingDone: true });
+  }
+
+  /**
+   * Launch pathfindings.
+   *
+   * For each path, launch a pathfinding request (one at the time).
+   * If a pathfinding is not successful, the corresponding trainSchedules will not be imported.
+   */
+  async function generatePaths(autocomplete = false) {
+    const pathErrors = [];
+    const pathFindings = {};
+    const pathFindingsNumber = pathsDictionnary.length;
+    for (let i = 0; i < pathFindingsNumber; i += 1) {
+      const path = pathsDictionnary[i];
+      const pathRef = path.trainNumber;
+      const { payload, rollingStockId } = generatePathfindingPayload(
+        trainsWithPathRef,
+        rollingStockDB,
+        path,
+        rollingStockID,
+        infraID,
+        autocomplete,
+        pointsDictionnary
```

Comment:
pathsDictionnary.forEach ?

Reply:
I need to keep this synchrone, that is why I used a loop here

Reply:
what do you mean ? forEach and a for loop are both synchrone